Src/LiquidProjections.PollingEventStore/PollingEventStoreAdapter.cs

```diff
@@ -81,50 +106,101 @@ public IDisposable Subscribe(long? lastProcessedCheckpoint, Subscriber subscribe
             }
 
+            LoadRequest pendingRequest =
+                pendingRequests.ToArray().SingleOrDefault(x => x.PrecedingCheckpoint == precedingCheckpoint);
+
+            if (pendingRequest != null)
+            {
+                logDebug(() =>
+                    $"Pending Request Found (subscription: {subscriptionId}, originalSubscription: " +
+                    $"{pendingRequest.SubscriptionId}, preceding checkpoint: {precedingCheckpoint}");
+
+                return await pendingRequest.Page;
+            }
 
-            Page loadedPage = await LoadNextPage(lastProcessedCheckpoint, subscriptionId).ConfigureAwait(false);
- 
-            if (loadedPage.Transactions.Count == maxPageSize)
+            var completedPage = await RequestPageload(subscriptionId, precedingCheckpoint, cancellationToken);
+            if (ShouldPreloadNextPage(completedPage))
             {
-                StartPreloadingNextPage(loadedPage.LastCheckpoint, subscriptionId);
+                // Ingore the result. It will become available through the queue.
+                Task _ = RequestPageload(subscriptionId, completedPage.PrecedingCheckpoint, cancellationToken);
             }
 
-            return loadedPage;
+            return completedPage;
+        }
+
+        private Task<Page> RequestPageload(string subscriptionId, long precedingCheckpoint, CancellationToken cancellationToken)
+        {
+            var loadRequest = new LoadRequest
+            {
+                SubscriptionId = subscriptionId,
+                PrecedingCheckpoint = precedingCheckpoint,
+                CancellationToken = cancellationToken
+            };
+            
+            pendingRequests.Enqueue(loadRequest);
+            requestQueued.TrySetResult(true);
+
+            return loadRequest.Page;
+        }
+
+        private bool ShouldPreloadNextPage(Page completedPage)
+        {
+            // NOTE: Only preload when caching is enabled. If it is not, other subscribers will only benefit from preloading if the request for a particular
+            // page is still in the queue.
+            return (cachedTransactionsByPrecedingCheckpoint != null) && (completedPage?.Transactions.Count == maxPageSize);
         }
 
         private Page TryGetNextPageFromCache(long precedingCheckpoint, string subscriptionId)
         {
-            if ((cachedTransactionsByPrecedingCheckpoint != null) && cachedTransactionsByPrecedingCheckpoint.TryGet(precedingCheckpoint, out Transaction cachedTransaction))
+            if ((cachedTransactionsByPrecedingCheckpoint != null) &&
+                cachedTransactionsByPrecedingCheckpoint.TryGet(precedingCheckpoint, out Transaction cachedTransaction))
             {
-                var resultPage = new List<Transaction>(maxPageSize) { cachedTransaction };
+                var resultPage = new List<Transaction>(maxPageSize) {cachedTransaction};
 
                 while (resultPage.Count < maxPageSize)

```

Comment:
❌ As discussed. If only the first N transactions of the requested page are in the cache, this will loop forever.

Reply:
Already fixed